File: src/test/java/ch/zxseitz/j3de/math/Vector2Test.java
package ch.zxseitz.j3de.math;

import org.junit.Assert;
import org.junit.Test;

public class Vector2Test {
    @Test
    public void testEquals() {
        var a = new Vector2(1f, 2f);
        var b = new Vector2(1f, 2f);

        Assert.assertEquals(a, b);
    }

    @Test
    public void testToString() {
        var a = new Vector2(1f, 2f);

        Assert.assertEquals("(1.000, 2.000)", a.toString());
    }

    @Test
    public void testAdd() {
        var a = new Vector2(1f, 2f);
        var b = new Vector2(3f, 4f);

        Assert.assertEquals(new Vector2(4f, 6f), a.add(b));
    }

    @Test
    public void testSubtract() {
        var a = new Vector2(1f, 2f);
        var b = new Vector2(3f, 4f);

        Assert.assertEquals(new Vector2(-2f, -2f), a.subtract(b));
    }

    @Test
    public void testMultiply() {
        var a = new Vector2(1f, 2f);
        var b = new Vector2(3f, 4f);

        Assert.assertEquals(new Vector2(3f, 8f), a.multiply(b));
    }

    @Test
    public void testDivide() {
        var a = new Vector2(1f, 2f);
        var b = new Vector2(3f, 4f);

        Assert.assertEquals(new Vector2(1f / 3f, 1f / 2f), a.divide(b));
    }

    @Test
    public void testScale() {
        var a = new Vector2(1f, 2f);
        var s = 3f;

        Assert.assertEquals(new Vector2(3f, 6f), a.scale(s));
    }

    @Test
    public void testNormSquared() {
        var a = new Vector2(1f, 2f);

        Assert.assertEquals(5f, a.normSquared(), MathUtils.TOLERANCE);
    }

    @Test
    public void testNorm() {
        var a = new Vector2(1f, 2f);

        Assert.assertEquals(2.2360679f, a.norm(), MathUtils.TOLERANCE);
    }

    @Test
    public void testDot() {
        var a = new Vector2(1f, 2f);
        var b = new Vector2(3f, 4f);

        Assert.assertEquals(11f, a.dot(b), MathUtils.TOLERANCE);
    }
}

File: src/test/java/ch/zxseitz/j3de/math/Vector2Test.kt
package ch.zxseitz.j3de.math

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Vector2Test {
    @Test
    fun testEquals() {
        val a = Vector2(1f, 2f)
        val b = Vector2(1f, 2f)
        assertEquals(a, b)
    }

    @Test
    fun testToString() {
        val a = Vector2(1f, 2f)
        assertEquals("(1.000, 2.000)", a.toString())
    }

    @Test
    fun testAdd() {
        val a = Vector2(1f, 2f)
        val b = Vector2(3f, 4f)
        assertEquals(Vector2(4f, 6f), a + b)
    }

    @Test
    fun testSubtract() {
        val a = Vector2(1f, 2f)
        val b = Vector2(3f, 4f)
        assertEquals(Vector2(-2f, -2f), a - b)
    }

    @Test
    fun testMultiply() {
        val a = Vector2(1f, 2f)
        val b = Vector2(3f, 4f)
        assertEquals(Vector2(3f, 8f), a _times b)
    }
    @Test
    fun testScale() {
        val a = Vector2(1f, 2f)
        val s = 3f
        assertEquals(Vector2(3f, 6f), a * s)
    }

    @Test
    fun testNormSquared() {
        val a = Vector2(1f, 2f)
        assertEquals(5f, a.normSquared(), MathUtils.TOLERANCE)
    }

    @Test
    fun testNorm() {
        val a = Vector2(1f, 2f)
        assertEquals(2.2360679f, a.norm(), MathUtils.TOLERANCE)
    }

    @Test
    fun testDot() {
        val a = Vector2(1f, 2f)
        val b = Vector2(3f, 4f)
        assertEquals(11f, a dot b, MathUtils.TOLERANCE)
    }
}
